### ISSUE TITLE
Consider shortening and adding a solutions branch the Node.js intro workshop

### ISSUE THREAD

**Issue by MissArray** [foundersandcoders/master-reference#921]
Launching into Node.js with a [three-hour code-along](https://github.com/foundersandcoders/Node-Intro-Workshop), followed by the even longer Nodegirls workshop is not ideal. The intro workshop doesn't contain a solutions branch and the current mentor walkthrough is fine, if you are a mentor and familiar with Node.js, but probably confusing if you're a newcomer to Node.

The Node.js introductory workshop is great if you're already familiar with Node, but, in my view, too demanding if you don't.

**Length**
Three hours is too much for a code-along, especially on a subject that's relatively new: on the basis of the current schedule, the students research Node, modules, project architecture etc. two days later, which seems somewhat counterintuitive. Having that knowledge would make it much easier for them to understand what's going on in both the introductory workshop and in the Nodegirls workshop.

Also, it can be hard to keep the same level of concentration three hours in a row. The sheer amount of code they need to type in the available time means that being distracted for even a few seconds is enough for students to miss some important step and get lost and confused.

**Modularisation**
The workshop itself would benefit, I think, from the principles of modularisation that it demonstrates. Aiming to 
1. show the advantages of Node.js, 
2. show how an HTTP server is built
3. show how a basic Node project is structured
4. refactor the code, with lots of cutting and pasting between files
in a single workshop may be a little too ambitious. This will depend on a cohort's overall understanding of and familiarity with code, of course, but for genuine newcomers, it may prove too much.

**No solutions branch**
Putting together a guide with an explanation for and a link to each step could make a real difference: students would be able to follow the steps, even if they get lost along the way, and having recourse to an overview of how set up a Node.js HTTP server could also help them prepare their projects.

**Comment by m4v15** [comment]
For the record in the West Bank this day actually went really well with good feedback from the students - however we had an extra hour in our day for teaching compared to London, so maybe this allowed us a bit more clarity in terms of the code along. Also we have a few mentors here, so keeping track of how people are keeping up in the code along is a bit easier - we have 2/3 people floating for 1 person delivering the code along.

For the delivery here, I did a short presentation that was effectively the README.md, and then @mattlub did the code along, with me + others floating to see what people are typing.

It did take a long time but feedback from the students was very positive - again possibly because of the reasons I mentioned above.

I think a solution branch **definitely** makes sense to add, although I guess the idea is that by the end the students should all have written the solution.

>refactor the code, with lots of cutting and pasting between files in a single workshop may be a little too ambitious. This will depend on a cohort's overall understanding of and familiarity with code, of course, but for genuine newcomers, it may prove too much.

Kind of agree, this is hard to code along as well with many windows. However it is definitely an important concept to cover so would be hesitant just to drop it.

One thing we thought about was the fact that the first half of node girls is very similar to this - maybe a potential solution is somehow combining them - code along up until a certain stage of NG and then leave them to continue themselves. However this loses the benefit of repetition which is obviously a helpful learning tool.

**Comment by MissArray** [comment]
Thanks for adding your feedback from the West Bank cohort, @m4v15. Yes, I am positive that having an extra hour is essential for introducing Node, pausing to explain in sufficient detail what happens at each step, allowing students to ask questions and catch up (and no doubt a fourth mentor would make a difference).

The time estimates for this workshop corroborate what you report.

I'm raising this issue on the basis of student feedback on this week's workshops at the SGC and on FAC14 comments on this particular workshop when we did it. I was reminded that one of the FAC14 was in tears during either that or the Nodegirls workshop, because she'd got so lost at the time.

Anyway; FAC15 know that at least two of them will come to deliver the same workshop, so it's also up to them to flag it and make suggestions for improving it, if they wish.

**Comment by mattlub** [comment]
@MissArray I agree it probably needs more time scheduled- as @m4v15 said we had more time, and also shortened step 3 (didn't serve the js or favicon) to save time.

> use the mentor walkthrough notes as the basis for a 'workshop walkthrough'

The mentor walkthrough notes could be fleshed out into more of a student-friendly walkthrough- however, wouldn't it just become more and more similar to the Node girls workshop (which is already a good resource)?

**Comment by MissArray** [comment]
@mattlub the 'workshop walkthrough' branch I'm suggesting would serve as a '**solutions**' branch; in short, students would be expected to consult it _after_ the workshop or if they got lost or found it hard to follow the logic of the steps. 

I agree that the Nodegirls workshop is a good resource; however, several in both FAC14 and FAC15 seemed to find it frustrating when they had to go through it. Quite interesting to see that what makes sense with the hindsight accumulated over the course can seem so opaque when you first come across it. I think that it's worth keeping this difference in mind when designing/reviewing workshops for FAC students.

In any case, FAC15 will be asked to review the curriculum at some point, as we were, so hopefully the points we've raised here will help them plan their first Node week - as mentors, that is. 

In the meantime, they can always raise their own issues on GitHub about any workshop/challenge they're not happy with. I did remind them that, in general, if they really feel that something in the curriculum needs to change, fulminating about it in an SGC is not enough; they need to raise an issue (we're already at #921; we should have a party or something when we hit #1000).
